java/code/src/com/suse/manager/webui/menu/MenuTree.java

```diff
@@ -231,8 +231,10 @@
 
             // Content Management
             nodes.add(new MenuItem("contentmanagement.nav.title").withIcon("spacewalk-icon-software-channels")
-                .addChild(new MenuItem("contentmanagement.nav.list")
-                        .withPrimaryUrl("/rhn/manager/contentmanagement/projects")));
+                    .withDir("/rhn/manager/contentmanagement")
```

Comment:
tl;dr : `withDir` is a fallback **partial** match of the URL that looks for an active menu item to select. I saw many times not having a fallback `withDir` reflected into *no-menu-item-active* result.

---

The `withDir` means *everything under this `path` that has no better specific detected URL MenuItem* it will be reflected in the menu as belonging to this `MenuItem` even if it does not match perfectly. It's like a wrapper to say *if you do not find anything that matches with the current URL, mark this MenuItem as selected as a fallback*. The same is happening also at [L237](https://github.com/uyuni-project/uyuni/pull/430/files/#diff-99972ff1af9cd70a9cfcb95a01374f18R237).

For instance:
- request is `/rhn/manager/contentmanagement/project/my-test-project`
- no **perfect** URL MenuItem is found
- the [`getBestActiveDir`](https://github.com/uyuni-project/uyuni/blob/dae038788ce3a63c92ae3d9d45cec11cebcecb7b/java/code/src/com/suse/manager/webui/menu/MenuTree.java#L502) method will look for the *best partial matching* MenuItem

In the `.withDir("/rhn/manager/contentmanagement/project")` case it will mark as `active` the `new MenuItem("contentmanagement.nav.projects")` element (second level) of the menu in any single Project page. In the `.withDir("/rhn/manager/contentmanagement/")` it will mark the `new MenuItem("contentmanagement.nav.title")` element (first level) of the menu. The latter I don't expect to happen but in case we will add any additional URL/Page without a specific MenuItem, we already have a fallback.